src/main/java/org/sagebionetworks/bridge/notification/worker/BridgeNotificationWorkerProcessor.java

```diff
@@ -0,0 +1,360 @@
+    private boolean shouldExcludeUser(String studyId, StudyParticipant participant) {
+        WorkerConfig workerConfig = dynamoHelper.getNotificationConfigForStudy(studyId);
+        Set<String> excludedDataGroupSet = workerConfig.getExcludedDataGroupSet();
+
+        // Unverified phone numbers can't be notified
+        if (Boolean.FALSE.equals(participant.getPhoneVerified())) {
+            return true;
+        }
+
+        // Users without timezones can't be processed
+        if (participant.getTimeZone() == null) {
+            return true;
+        }
+
+        // Users with timezone < UTC-11 or > UTC-1 should be excluded. This is because we'd end up sending at unusually
+        // early or unusually late hours.
+        DateTimeZone timeZone = DateUtils.parseZoneFromOffsetString(participant.getTimeZone());
+        int timeZoneOffsetMillis = timeZone.getOffset(DateTime.now());
+        if (timeZoneOffsetMillis < MIN_TIMEZONE_OFFSET_MILLIS || timeZoneOffsetMillis > MAX_TIMEZONE_OFFSET_MILLIS) {
+            return true;
+        }
+
+        // Unconsented users can't be notified
+        if (!isUserConsented(studyId, participant)) {
+            return true;
+        }
+
+        // If the user has any of the excluded data groups, exclude the user
+        for (String oneUserDataGroup : participant.getDataGroups()) {
+            if (excludedDataGroupSet.contains(oneUserDataGroup)) {
+                return true;
+            }
+        }
+
+        // If user was already sent a notification in the last burst duration, don't send another one
+        Long notificationTime = dynamoHelper.getLastNotificationTimeForUser(participant.getId());
+        if (notificationTime != null && notificationTime > DateTime.now()
+                .minusDays(workerConfig.getBurstDurationDays()).getMillis()) {
+            return true;
+        }
+
+        // We've checked all the exclude conditions. Do not exclude user.
+        return false;
+    }
+
+    // Helper method to determine if the user has signed the required consent(s).
+    private boolean isUserConsented(String studyId, StudyParticipant participant) {
+        // Check each required subpop. The user must have signed that consent.
+        WorkerConfig workerConfig = dynamoHelper.getNotificationConfigForStudy(studyId);
+        Map<String, List<UserConsentHistory>> consentsBySubpop = participant.getConsentHistories();
+        for (String oneRequiredSubpopGuid : workerConfig.getRequiredSubpopulationGuidSet()) {
+            List<UserConsentHistory> oneConsentList = consentsBySubpop.get(oneRequiredSubpopGuid);
+            if (oneConsentList.isEmpty()) {
+                return false;
+            }
+
+            // Newest consent is always at the end.
+            UserConsentHistory newestConsent = oneConsentList.get(oneConsentList.size() - 1);
+            if (!newestConsent.getHasSignedActiveConsent() || newestConsent.getWithdrewOn() != null) {
```

Comment:
I don't think they'll want to exclude people who have not signed the most recent consent, not unless they implement UI to have people sign an updated consent, and no one has every implemented that logic. More likely is that the study team fixes and publishes a typo, then this would be false.